syndesis-maven-plugin/src/main/java/io/syndesis/maven/GenerateMapperInspectionsMojo.java

```diff
@@ -0,0 +1,128 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.maven;
+
+import io.atlasmap.maven.GenerateInspectionsMojo;
+import io.syndesis.dao.init.ModelData;
+import io.syndesis.dao.init.ReadApiClientData;
+import io.syndesis.model.Kind;
+import io.syndesis.model.connection.Action;
+import io.syndesis.model.connection.Connector;
+import io.syndesis.model.connection.DataShape;
+import org.apache.maven.model.Resource;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+
+@Mojo(name = "generate-mapper-inspections", defaultPhase = LifecyclePhase.GENERATE_RESOURCES)
+public class GenerateMapperInspectionsMojo extends AbstractMojo {
+
+    @Parameter(readonly = true, defaultValue = "${project}")
+    private MavenProject project;
+
+    @Component
+    private RepositorySystem system;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true)
+    private List<RemoteRepository> remoteRepos;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    private RepositorySystemSession repoSession;
+
+    @Parameter(defaultValue = "${project.build.directory}/generated-sources/atlasmap")
+    private File outputDir;
+
+    @Parameter(defaultValue = "static/mapper/v1/java-inspections")
+    private String resourceDir;
+
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        try {
+
+            Resource resource = new Resource();
+            resource.setDirectory(outputDir.getCanonicalPath());
+            project.addResource(resource);
+
+            HashSet<File> generated = new HashSet<>();
+
+            ReadApiClientData reader = new ReadApiClientData();
+            List<ModelData<?>> modelList = reader.readDataFromFile("io/syndesis/dao/deployment.json");
```

Comment:
I'm not sure I agree with that.  We would end up duplicating configuration and that would lead to increased chances of getting it wrong.  By using the deployments.json as the source of truth, the generated pre-computations will always be accurate.